ProcessLib/Process.h

```diff
@@ -72,14 +72,59 @@ class Process
 	/// Process specific initialization called by initialize().
 	virtual void createLocalAssemblers() = 0;
 
-	/// Postprocessing after solve().
+	/// Preprocessing before starting assembly for new timestep.
+	virtual void preTimestep(GlobalVector const& x) {}
```

Comment:
Thanks for explanation. I understand that `preTimestep()` and `postTimestep()` should be able change internal things. Will `output()` change internals, too?